temporalio/converter.py

```diff
@@ -1151,6 +1151,7 @@ def encode_search_attributes(
             DEPRECATED.
         api: API message to set converted attributes on.
     """
+    api.indexed_fields.clear()  # Ensure that we at least create an empty map
```

Comment:
@cretz let's double-check we both agree about the change on this line. Without it, if the user sets an empty but non-None value, the request proto did not contain a `search_attributes` field at all. But what we want in that case is an empty map, which clears all SAs server-side, as per the [spec](https://github.com/temporalio/api/blob/master/temporal/api/workflowservice/v1/request_response.proto#L1186-L1187) for this feature, and in agreement with e.g. the Go tests.

Of course this line affects all other code paths, and I have also made it so that it affects both the untyped- as well as typed- search attributes, even though I only have evidence of a bug in the case of typed SAs when used with the UpdateSchedule request.